Xamarin.Forms.Core/Segues/NavigationAction.cs

```diff
@@ -0,0 +1,80 @@
+﻿using System;
+
+using Xamarin.Forms;
+
+namespace Xamarin.Forms
+{
+	[TypeConverter(typeof(NavigationActionConverter))]
+	public enum NavigationAction
```

Comment:
Because the `PopX` are `OR`ed together, shouldn't they all be powers of two (and be `[Flags]`)? Although as it turns out everything will be fine in this case with them having values of 5, 6, 7 respectively.

Since the `PopX` values are special, and `NavigationAction`s can't arbitrarily be `OR`ed anyway, it seems better to specifically assign values to them.